qiskit/opflow/list_ops/list_op.py

```diff
@@ -101,6 +101,11 @@ def _state(
             "grad_combo_fn": grad_combo_fn if grad_combo_fn is not None else self.grad_combo_fn,
         }
 
+    @property
+    def settings(self) -> Dict:
+        """Return settings."""
+        return {"oplist": self._oplist, "coeff": self._coeff, "abelian": self._abelian}
+
```

Comment:
I see that sub-classes like SummedOp etc use this. But should this check if self is of ListOp type (not a sub-class) and raise an exception if combo_fn is not the default one. And also check/exception if grad_combo_fn is not None.

I see oplist is just the list itself. It does not recurse here to generate nested dict of settings then by calling settings on each of the contained objects in the list - just returns it raw so to speak. All the OperatorBase objects in that list I guess are handled somehow externally

Reply:
Also fixed in 0dfcc3b. 

There is no recursion here because `settings` is created for JSON encoder, which already does the recursion.

Reply:
I do not know the JSON encoder that well - is `settings` something it knows about and people would be familiar with. When I look  at the class and see `settings` its not so clear what its intent is - especially as list op contains OperatorBase instances which all have settings too. Maybe the base class should at least explain a bit further what the intent of this property is as regards what it should be/contain - since saying it returns settings as docstring says nothing more beyond what is self-evident from the code thats its a getter of settings.

Reply:
`settings` is not a standard name to use by JSON encoder. It's just a name @Cryoris and @mtreinish  came up with that sounds better than `to_dict()`. 

The docstring was intentionally vague because right now it's only used by `ibmq-provider` to serialize the object, but we don't want to make it `ibmq` specific. Having said that,  in bc204ea I updated the docstring in `OperatorBase` to say _one_ of the use cases is to serialize the object.

Reply:
Ok, I guess this just represents the properties belong to that object - even if an object in the list is/or has other objects that have settings it seems we do not not have nested settings dicts here. It was more that if we add more opflow objects knowing what to do/expect in regards of that method implementation. Although I think looking at the PR you referenced the ones that can be serialized are also named somewhere so a new class here, even if it had settings, I guess does not get serialized directly. If its part of a ListOp, which is taken care of by the JSON Encoder you say, maybe it would be serialized in that case.

Reply:
> Ok, I guess this just represents the properties belong to that object - even if an object in the list is/or has other objects that have settings it seems we do not not have nested settings dicts here

Yeah, perhaps that's why they picked `settings` instead of `to_dict()`, since the latter does convert all nested objects to dictionaries and the former doesn't.

> Although I think looking at the PR you referenced the ones that can be serialized are also named somewhere so a new class here, even if it had settings, I guess does not get serialized directly.

I just updated the `ibmq-provider` PR to make it more general. It now always uses `settings` if the object has one, instead of using a lookup table, so new classes will get serialized automatically.